index.js

```diff
@@ -46,20 +46,28 @@ exports.Exporter = class {
   async getLastPosition() {
     if (await zookeeperClient.existsAsync(this.zookeeperPositionNode)) {
       const previousBlockNumber = await zookeeperClient.getDataAsync(this.zookeeperPositionNode)
-      return previousBlockNumber.data.readUInt32BE(0)
-    } else {
-      return null
+
+      if (previousBlockNumber && Buffer.isBuffer(previousBlockNumber.data)) {
+        try {
+          return JSON.parse(previousBlockNumber.data.toString('utf8'))
```

Comment:
I think that won’t be backward compatible with the old API that works with numbers. Can we make it smart and continue supporting the old format?

Reply:
Do you have an idea how to "detect" if a value is in the old format?

Reply:
Added a format header to detect new values